tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -964,15 +967,19 @@ void test_tls_leakage(void)
 
 	k_thread_user_mode_enter(tls_leakage_user_part,
 				 _current->userspace_local_data, NULL, NULL);
+#endif
 }
 
 void tls_entry(void *p1, void *p2, void *p3)
 {
+#ifdef CONFIG_THREAD_USERSPACE_LOCAL_DATA
 	printk("tls_entry\n");
+#endif
 }
```

Comment:
would be nice to place `#ifdef CONFIG_THREAD_USERSPACE_LOCAL_DATA` for the entire function, unless there is a reason we cannot do that.

Reply:
Yup, looks like I approached this problem incrementally :-)